src/devices/zigbeetlc.ts

```diff
@@ -0,0 +1,246 @@
+            {modelID: 'MHO-C401N-bz', manufacturerName: 'MiaoMiaoCe', model: 'MHO-C401N'},
+        ],
+        model: 'LYWSD03MMC',
+        vendor: 'Xiaomi',
+        description: 'Temp & RH Monitor Lite (pvxx/ZigbeeTLc)',
+        fromZigbee: [fz.temperature, fz.humidity, fz.battery],
+        toZigbee: [tz.thermostat_temperature_display_mode],
+        exposes: [
+            e.battery(), e.temperature(), e.humidity(),
+            e.enum('temperature_display_mode', ea.ALL, ['celsius', 'fahrenheit'])
+                .withDescription('The temperature unit displayed on the screen'),
+        ],
+        extend: [
+            extend.endpointQuirk,
+            extend.tempCalibration,
+            extend.humidityCalibration,
+            extend.measurementInterval,
+            extend.display,
+            extend.comfortDisplay,
+            extend.comfortTemperatureMin,
+            extend.comfortTemperatureMax,
+            extend.comfortHumidityMin,
+            extend.comfortHumidityMax,
+        ],
+        ota: ota.zigbeeOTA,
+        configure: configure,
+    },
+    /*
+        ZigbeeTLc devices supporting:
+        - Temperature (+calibration)
+        - Humidity (+calibration)
+        - Display
+    */
+    {
+        fingerprint: [
+            {modelID: 'CGDK2-z', manufacturerName: 'Qingping', model: 'CGDK2'},
+            {modelID: 'CGDK2-bz', manufacturerName: 'Qingping', model: 'CGDK2'},
+        ],
+        model: 'CGDK2',
+        vendor: 'Qingping',
+        description: 'Temp & RH Monitor Lite (pvxx/ZigbeeTLc)',
+        fromZigbee: [fz.temperature, fz.humidity, fz.battery],
+        toZigbee: [tz.thermostat_temperature_display_mode],
+        exposes: [
+            e.battery(), e.temperature(), e.humidity(),
+            e.enum('temperature_display_mode', ea.ALL, ['celsius', 'fahrenheit'])
+                .withDescription('The temperature unit displayed on the screen'),
+        ],
+        extend: [
+            extend.endpointQuirk,
+            extend.tempCalibration,
+            extend.humidityCalibration,
+            extend.display,
+            extend.measurementInterval,
+        ],
+        ota: ota.zigbeeOTA,
+        configure: configure,
+    },
+    /*
+        ZigbeeTLc devices supporting:
+        - Temperature (+calibration)
+        - Humidity (+calibration)
+    */
+    {
+        fingerprint: [
+            {modelID: 'TS0201-z', manufacturerName: 'Tuya', model: 'TS0201'},
+            {modelID: 'TS0201-bz', manufacturerName: 'Tuya', model: 'TS0201'},
+            {modelID: 'TH03Z-z', manufacturerName: 'Tuya', model: 'TH03Z'},
+            {modelID: 'TH03Z-bz', manufacturerName: 'Tuya', model: 'TH03Z'},
+        ],
+        model: 'WSD500A',
+        vendor: 'Tuya',
+        description: 'Temperature & Humidity Sensor (pvxx/ZigbeeTLc)',
+        fromZigbee: [fz.temperature, fz.humidity, fz.battery],
+        toZigbee: [],
+        exposes: [e.battery(), e.temperature(), e.humidity()],
```

Comment:
Please use the `modernExtend` version instead. I guess you did not use them because some of the attributes might fail to setup? What is the error thrown in this case? I think we should add a property `allowError` to the modernExtends to be able to ignore it.